crates/ruff_linter/src/rules/flake8_simplify/rules/zip_dict_keys_and_values.rs

```diff
@@ -40,17 +40,19 @@ pub struct ZipDictKeysAndValues {
 impl AlwaysFixableViolation for ZipDictKeysAndValues {
     #[derive_message_formats]
     fn message(&self) -> String {
-        let ZipDictKeysAndValues { expected, actual } = self;
-        if let (Some(expected), Some(actual)) = (expected.full_display(), actual.full_display()) {
+        if let (Some(expected), Some(actual)) =
+            (self.expected.full_display(), self.actual.full_display())
+        {
             format!("Use `{expected}` instead of `{actual}`")
         } else {
-            format!("Use `dict.items()` instead of `zip(dict.keys(), dict.values())`")
+            "Use `dict.items()` instead of `zip(dict.keys(), dict.values())`".to_string()
         }
     }
 
     fn fix_title(&self) -> String {
-        let ZipDictKeysAndValues { expected, actual } = self;
-        if let (Some(expected), Some(actual)) = (expected.full_display(), actual.full_display()) {
+        if let (Some(expected), Some(actual)) =
+            (self.expected.full_display(), self.actual.full_display())
+        {
```

Comment:
Same here, the unpacking of `self` was fine and resulted in fewer lines of code overall. I'd keep the unpacking as it was in these places.